Consume ENV vars where possible

mkolesnik opened the pull request:
Instead of consuming flags, consume environment variables where
possible.
Some flags are retained for backwards compatibility not to break
consumers, and will be removed once their consumers are updated.

This is the first step in resolving #887 

Depends on #886 

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

mkolesnik committed:
Consume ENV vars for core deploy scripts

Instead of consuming flags, consume environment variables where
possible.
Some flags are retained for backwards compatibility not to break
consumers, and will be removed once their consumers are updated.

This commit handles the core scripts (cleanup, clusters, deploy, e2e) in
order to have a manageable review process.

Bumped CI size check to accommodate this commit, as all these scripts
are inter-connected and as such would be harder to refactor in
piece-meal fashion.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

mkolesnik committed:
Consume ENV vars for auxiliary scripts

Instead of consuming flags, consume environment variables where
possible.
Some flags are retained for backwards compatibility not to break
consumers, and will be removed once their consumers are updated.

This commit handles the auxiliary scripts (compile, images, release)
in order to have a manageable review process.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

mkolesnik committed:
Drop all caching logic from image building

Since we're using docker BuildKit which handles caching quite well on
its own, theres no need to try and force use of cached layers anymore.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

skitt commented on .github/workflows/linting.yml:
```diff
@@ -40,7 +40,7 @@ jobs:
         uses: ./gh-actions/commit-size
         with:
-          size: 200
+          size: 250
```
If this gets updated for any reason, could you mention this change in the corresponding commit message?

skitt commented on Makefile.inc:
```diff
@@ -31,73 +31,58 @@ include $(SHIPYARD_DIR)/Makefile.versions
 
 ifneq (,$(filter libreswan,$(_using)))
-override DEPLOY_ARGS += --cable_driver libreswan
+export CABLE_DRIVER = libreswan
```
Since these are exported further down, how about

    CABLE_DRIVER = lighthouse

Same for the other variables.

mkolesnik committed:
Add and use standard print in all scripts

Instead of using a different (or no) print, use the same print for all
script in order to make it clear which variables the script is running
with.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

skitt commented on scripts/shared/lib/utils:
```diff
@@ -189,3 +189,10 @@ function load_settings_from_yaml() {
+
+function print_env() {
+   echo "Running with environment variables:"
+   for var; do
+       eval echo "${var}=\${${var}@Q}"
+   done
```
The safe way of doing this is

```
    for var; do
        declare -n value=$var
        printf '%s=%q\n' "$var" "$value"
    done
```

mkolesnik replied:
Cool, good to know :D

skitt reviewed (changes requested):
Regarding the cache-removing commit, note that we only use `buildx` for multi-arch images.

mkolesnik commented:
> Regarding the cache-removing commit, note that we only use `buildx` for multi-arch images.

Yes you're right of course, it's BuildKit that has the improvements (well probably buildx too as BuildKit is it's backed).
I updated the commit message.
